src/AWS.Deploy.Orchestration/CustomRecipeLocator.cs

```diff
@@ -0,0 +1,185 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using AWS.Deploy.Common.DeploymentManifest;
+using AWS.Deploy.Common.IO;
+using AWS.Deploy.Orchestration.Utilities;
+
+namespace AWS.Deploy.Orchestration
+{
+    public interface ICustomRecipeLocator
+    {
+        Task<HashSet<string>> LocateCustomRecipePaths(string targetApplicationFullPath, string solutionDirectoryPath);
+    }
+
+    /// <summary>
+    /// This class supports the functionality to fetch custom recipe paths from a deployment-manifest file as well as
+    /// other locations that are monitored by the same source control root as the target application that needs to be deployed.
+    /// </summary>
+    public class CustomRecipeLocator : ICustomRecipeLocator
+    {
+        private const string GIT_STATUS_COMMAND = "git status";
+        private const string SVN_STATUS_COMMAND = "svn status";
```

Comment:
Discussed offline, the usage of git/svn commands is source dependent rather than artifacts generated on building.

Not ship blocking.